src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -405,8 +390,137 @@ void RegisterAppInterfaceRequest::Run() {
   }
 
   GetPolicyHandler().SetDeviceInfo(device_mac, device_info);
+}
+
+RegisterAppInterfaceRequest::DataResumeResult
+RegisterAppInterfaceRequest::ApplicationDataShouldBeResumed() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const auto& msg_params = (*message_)[strings::msg_params];
+  resumption::ResumeCtrl& resume_ctrl =
+      application_manager_.resume_controller();
+
+  bool resumption = msg_params.keyExists(strings::hash_id);
+  if (!resumption) {
+    LOG4CXX_DEBUG(logger_, "Hash id is missing, no resumption required");
+    return DataResumeResult::NO_HASH;
+  }
+  ApplicationSharedPtr application =
+      application_manager_.application(connection_key());
+  const auto& hash_id = msg_params[strings::hash_id].asString();
+  if (!resume_ctrl.CheckApplicationHash(application, hash_id)) {
+    LOG4CXX_DEBUG(logger_, "Hash from RAI does not match to saved resume data.");
+    return DataResumeResult::WRONG_HASH;
+  }
+  DCHECK(application);
+  if (!resume_ctrl.CheckPersistenceFilesForResumption(application)) {
+    LOG4CXX_DEBUG(logger_,
+                 "Persistent data is missing for "
+                     << application->policy_app_id());
+    return DataResumeResult::MISSED_DATA;
+  }
+  return DataResumeResult::RESUME_DATA;
+}
+
+void RegisterAppInterfaceRequest::Run() {
+  using namespace helpers;
+  LOG4CXX_AUTO_TRACE(logger_);
+  LOG4CXX_DEBUG(logger_, "Connection key is " << connection_key());
+
+  WaitForHMIReady();
+
+  if (application_manager_.IsStopping()) {
+    LOG4CXX_WARN(logger_, "The ApplicationManager is stopping!");
+    return;
+  }
+
+  if (ProcessApplicationTransportSwitching()) {
+    return;
```

Comment:
@LuxoftAKutsan suggest to add log messages for all `if` with `return` to understand what was the reason